soroban-env-host/src/storage.rs

```diff
@@ -476,3 +476,19 @@ impl Storage {
         Ok(())
     }
 }
+
+#[cfg(any(test, feature = "testutils"))]
+impl Storage {
+    /// Unmetered function to access entry with lifetimes to test lifetime extensions.
+    pub fn get_entry_with_lifetime(
+        &self,
+        key: Rc<LedgerKey>,
+    ) -> Result<Option<EntryWithLiveUntil>, HostError> {
```

Comment:
The tests seem to be only interested in lifetime... So why not make this to just return lifetime?

Reply:
Yes, the test only needs the lifetime, but since there seems to be no other way to do an unmetered get on an entry from the host this might be used in the future. So it's just an addition to allow for more test customizability in the future, but can for sure reshape it to return the lifetime, let me know.